plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/security/KafkaSecurityConfig.java

```diff
@@ -0,0 +1,167 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.kafka.security;
+
+import com.google.common.collect.ImmutableMap;
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.configuration.ConfigSecuritySensitive;
+import org.apache.kafka.common.config.SslConfigs;
+
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ *  {@SecurityConfig} manages Kafka SSL authentication and encryption between clients and brokers.
+ */
+public class KafkaSecurityConfig
+{
+    private String keyPassword;
+    private String keystoreLocation;
+    private String keystorePassword;
+    private KafkaKeystoreTruststoreType keystoreType;
+    private String truststoreLocation;
+    private String truststorePassword;
+    private KafkaKeystoreTruststoreType truststoreType;
+    private KafkaEndpointIdentificationAlgorithm endpointIdentificationAlgorithm;
+
+    public Optional<String> getKeyPassword()
+    {
+        return Optional.ofNullable(keyPassword);
+    }
+
+    @Config("kafka.ssl.key.password")
+    @ConfigDescription(SslConfigs.SSL_KEY_PASSWORD_DOC)
+    @ConfigSecuritySensitive
+    public KafkaSecurityConfig setKeyPassword(String keyPassword)
+    {
+        this.keyPassword = keyPassword;
+        return this;
+    }
+
+    public Optional<String> getKeystoreLocation()
+    {
+        return Optional.ofNullable(keystoreLocation);
+    }
+
+    @Config("kafka.ssl.keystore.location")
+    @ConfigDescription(SslConfigs.SSL_KEYSTORE_LOCATION_DOC)
+    public KafkaSecurityConfig setKeystoreLocation(String keystoreLocation)
+    {
+        this.keystoreLocation = keystoreLocation;
+        return this;
+    }
+
+    public Optional<String> getKeystorePassword()
+    {
+        return Optional.ofNullable(keystorePassword);
+    }
+
+    @Config("kafka.ssl.keystore.password")
+    @ConfigDescription(SslConfigs.SSL_KEYSTORE_PASSWORD_DOC)
+    @ConfigSecuritySensitive
+    public KafkaSecurityConfig setKeystorePassword(String keystorePassword)
+    {
+        this.keystorePassword = keystorePassword;
+        return this;
+    }
+
+    public Optional<KafkaKeystoreTruststoreType> getKeystoreType()
+    {
+        return Optional.ofNullable(keystoreType);
+    }
+
+    @Config("kafka.ssl.keystore.type")
+    @ConfigDescription(SslConfigs.SSL_KEYSTORE_TYPE_DOC)
+    public KafkaSecurityConfig setKeystoreType(KafkaKeystoreTruststoreType keystoreType)
+    {
+        this.keystoreType = keystoreType;
+        return this;
+    }
+
+    public Optional<String> getTruststoreLocation()
```

Comment:
Please use `Optional<@FileExists String>` and please add tests to `TestKafkaPlugin` that plugin is unable to load if one gives a file location that is incorrect.

Please do the same for other location properties.